autonomy_node:
  ros__parameters:
    # -----------------------------
    # main
    # -----------------------------
    tick_period: 0.005
    cmd_vel_topic: "/cmd_vel"
    imu_topic: "/imu/data_raw"
    euler_axes: "sxyz"

    # choose: "pd" or "lqr"
    balance_mode: "pd"

    # -----------------------------
    # feedforward (stand / rocking)
    # -----------------------------
    feedforward:
      control_hz: 100.0
      v_peak: 3.7
      forward_time: 0.10
      backward_time: 0.30
      ramp_time: 0.05
      yaw_rate: 0.0

      # pitch trigger
      use_pitch_trigger: true
      pitch_trigger_deg: 12.0
      pitch_trigger_mode: "either"   # either | positive | negative
      pitch_hold_s: 0.10

    # -----------------------------
    # balance: PD gains (used when balance_mode: "pd")
    # -----------------------------
    pdbalance:
      kp_pitch: 5.0          # [m/s per rad]
      kd_pitch: 0.10         # [m/s per (rad/s)]
      v_limit: 0.50          # [m/s]
      wz_cmd: 0.0            # keep yaw still
      pitch_soft_deg: 35.0   # safety soft stop (deg)

    # -----------------------------
    # balance: LQR params (used when balance_mode: "lqr")
    # -----------------------------
    lqrbalance:
      a: 12.0
      b: -2.0
      c: 20.0

      q_pitch: 25.0
      q_pitch_rate: 2.0
      r_vx: 1.0

      v_limit: 0.50
      wz_cmd: 0.0
      pitch_soft_deg: 35.0

      # sign convention (แก้ทิศทาง IMU/เฟรม)
      pitch_sign: 1.0
      pitch_rate_sign: 1.0
      vx_sign: 1.0
